java/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/RunTasks.java

```diff
@@ -62,40 +83,92 @@ public class RunTasks {
         if (type.equals("compaction")) {
             this.sqsJobQueueUrl = instanceProperties.get(COMPACTION_JOB_QUEUE_URL);
             this.clusterName = instanceProperties.get(COMPACTION_CLUSTER);
             this.containerName = COMPACTION_CONTAINER_NAME;
-            this.taskDefinition = instanceProperties.get(COMPACTION_TASK_DEFINITION_FAMILY);
+            this.fargateTaskDefinition = instanceProperties.get(COMPACTION_TASK_FARGATE_DEFINITION_FAMILY);
+            this.ec2TaskDefinition = instanceProperties.get(COMPACTION_TASK_EC2_DEFINITION_FAMILY);
+            autoScalingGroupName = instanceProperties.get(COMPACTION_AUTO_SCALING_GROUP);
         } else if (type.equals("splittingcompaction")) {
             this.sqsJobQueueUrl = instanceProperties.get(SPLITTING_COMPACTION_JOB_QUEUE_URL);
             this.clusterName = instanceProperties.get(SPLITTING_COMPACTION_CLUSTER);
             this.containerName = SPLITTING_COMPACTION_CONTAINER_NAME;
-            this.taskDefinition = instanceProperties.get(SPLITTING_COMPACTION_TASK_DEFINITION_FAMILY);
+            this.fargateTaskDefinition = instanceProperties.get(SPLITTING_COMPACTION_TASK_FARGATE_DEFINITION_FAMILY);
+            this.ec2TaskDefinition = instanceProperties.get(SPLITTING_COMPACTION_TASK_EC2_DEFINITION_FAMILY);
+            autoScalingGroupName = instanceProperties.get(SPLITTING_COMPACTION_AUTO_SCALING_GROUP);
         } else {
             throw new RuntimeException("type should be 'compaction' or 'splittingcompaction'");
         }
         this.maximumRunningTasks = instanceProperties.getInt(MAXIMUM_CONCURRENT_COMPACTION_TASKS);
         this.subnet = instanceProperties.get(SUBNET);
         this.fargateVersion = instanceProperties.get(FARGATE_VERSION);
+        this.launchType = instanceProperties.get(COMPACTION_ECS_LAUNCHTYPE);
+
+        String architecture = instanceProperties.get(COMPACTION_TASK_CPU_ARCHITECTURE).toUpperCase(Locale.ROOT);
+        Pair<Integer, Integer> requirements = getCpuMemoryForArch(architecture, launchType, instanceProperties);
+
+        this.scaler = new Scaler(asClient, ecsClient, autoScalingGroupName, this.clusterName,
+                requirements.getLeft(),
+                requirements.getRight(),
+                Duration.ofSeconds(instanceProperties.getInt(COMPACTION_EC2_SCALING_GRACE_PERIOD)));
+    }
+
+    /**
+     * Retrieves architecture specific CPU and memory requirements. This returns
+     * a triple containing the CPU requirement in the left element and memory
+     * requirement in the right element.
+     *
+     * @param architecture CPU architecture
+     * @param launchType the container launch type
+     * @param instanceProperties Sleeper instance properties
+     * @return CPU and memory requirements as per the CPU architecture
+     */
+    private static Pair<Integer, Integer> getCpuMemoryForArch(String architecture,
+            String launchType,
+            InstanceProperties instanceProperties) {
```

Comment:
Could this be extracted to ensure the same code is used for the CDK? Maybe it could be in the configuration module?

Reply:
Done

Reply:
Done